packages/expo-dev-launcher/ios/EXDevLauncherController.m

```diff
@@ -360,8 +345,24 @@ - (void)loadApp:(NSURL *)expoUrl onSuccess:(void (^ _Nullable)(void))onSuccess o
       if (manifest) {
         launchExpoApp(self->_updatesInterface.launchAssetURL, [EXManifestsManifestFactory manifestForManifestJSON:manifest]);
       }
-    } error:onError];
-  } onError:onError];
+    } error:^(NSError * _Nonnull error) {
+      if (@available(iOS 14, *)) {
+        // Try to retry if the network connection was rejected because of the luck of the lan network permission.
+        static BOOL shouldRetry = true;
+        NSString *host = expoUrl.host;
+
+        if (shouldRetry && ([host hasPrefix:@"192.168."] || [host hasPrefix:@"172."] || [host hasPrefix:@"10."])) {
```

Comment:
should we add `localhost` to this list, or are simulators not affected by this permission issue?

Reply:
simulators are not affected by this permission issue ;) So we don't have to do it.